src/main/java/io/temporal/internal/replay/ReplayWorkflowExecutor.java

```diff
@@ -115,14 +118,14 @@
           "First event in the history is not WorkflowExecutionStarted");
     }
     startedEvent = firstEvent.getWorkflowExecutionStartedEventAttributes();
-    wfStartTimeNanos = firstEvent.getTimestamp();
+    wfStartTimeNanos = Timestamps.toNanos(firstEvent.getEventTime());
 
     context =
         new ReplayWorkflowContextImpl(
```

Comment:
Should we change ReplayWorkflowContextImp to take in a java.time.Duration instead of the start time in Nanoseconds in a subsequent PR?

Reply:
I don't think Duration is the right class for timestamps which represent point in time.

Reply:
sorry I meant Timestamp. I definitely interchanged those quite a bit while doing this diff.